macros/dmt_unit_test.sql

```diff
@@ -1,9 +1,50 @@
-{% test unit_test(model, input_mapping, expected_output, name, description, compare_columns, depends_on) %}
-    {% set test_sql = dbt_datamocktool.get_unit_test_sql(model, input_mapping, depends_on)|trim %}
-    {% do return(dbt_utils.test_equality(expected_output, compare_model=test_sql, compare_columns=compare_columns)) %}
+{% test unit_test(model, input_mapping, expected_output, name, description, compare_columns, depends_on, test_case_list = []) %}
+    {# Support iterating through list of test cases #}
+    {% if test_case_list %}
+    {% set error_count = namespace(value=0) %}
+      {% for test_case in test_case_list %}        
+        {# String substitution for inputs #}
+        {% set individual_input_mapping = dict() %}
+        {% for k, v in input_mapping.items() %}
+          {# String substitution on the templated value #}
+          {% set templated_value = v|replace('@', test_case)|replace('\\', '') %}
+          {# Update copy of dictionary #}
+          {% do individual_input_mapping.update({k: render('{{' ~ templated_value ~ '}}')}) %}
+        {% endfor %}
+
+        {# String substitution for expected output #}
+        {# Equality test expects a Relation #}
+        {% set full_path = render('{{' ~ expected_output|replace('@', test_case)|replace('\\', '') ~ '}}') %}
+        {% set full_path_list = full_path.split('.') %}
+        {% set individual_expected_output = adapter.get_relation(*full_path_list) %}
+
+        {# Retrieve the SQL code with the input mapping applied, using mocked input #}
+        {% set individual_test_sql = dbt_datamocktool.get_unit_test_sql(model, individual_input_mapping, depends_on, test_case) %}
```

Comment:
```suggestion
        {% set individual_test_sql = dbt_datamocktool.get_unit_test_sql(model, individual_input_mapping, depends_on) %}
```
I think you need to get rid of `test_case` here, right? Or am I missing what it's supposed to be doing?

I get an error when running:
```
00:39:26  Encountered an error:
Compilation Error in test dbt_datamocktool_unit_test_stg_customers__ref_dmt__expected_stg_customers_____ref_dmt__raw_customers____1__2 (models/staging/schema.yml)
  macro 'dbt_macro__get_unit_test_sql' takes not more than 3 argument(s)
  
  > in macro test_unit_test (macros/dmt_unit_test.sql)
  > called by macro get_unit_test_sql (macros/dmt_get_test_sql.sql)
  > called by test dbt_datamocktool_unit_test_stg_customers__ref_dmt__expected_stg_customers_____ref_dmt__raw_customers____1__2 (models/staging/schema.yml)
```